zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -0,0 +1,970 @@
+        // range on 8 bits.
+        let range_chip = RangeChip::new(config.range_config.clone(), 8);
+        let mut ecc_chip = GeneralEccChip::<Secp256k1Affine, F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>::new(
+            config.ecc_chip_config(),
+        );
+        let scalar_chip = ecc_chip.scalar_field_chip();
+
+        layouter.assign_region(
+            || "ecc chip aux",
+            |mut region| self.assign_aux(&mut region, &mut ecc_chip),
+        )?;
+
+        let ecdsa_chip = EcdsaChip::new(ecc_chip.clone());
+        let address_is_zero_chip = IsZeroChip::construct(config.address_is_zero.clone());
+
+        let mut assigned_ecdsas = Vec::new();
+        let mut keccak_auxs = Vec::new();
+
+        let chips = ChipsRef {
+            main_gate: &main_gate,
+            range_chip: &range_chip,
+            ecc_chip: &ecc_chip,
+            scalar_chip: &scalar_chip,
+            ecdsa_chip: &ecdsa_chip,
+        };
+
+        layouter.assign_region(
+            || "ecdsa chip verification",
+            |mut region| {
+                assigned_ecdsas.clear();
+                keccak_auxs.clear();
+                let offset = &mut 0;
+                let mut ctx = RegionCtx::new(&mut region, offset);
+                for i in 0..MAX_VERIF {
+                    let signature = if i < signatures.len() {
+                        signatures[i].clone()
+                    } else {
+                        // padding (enabled when address == 0)
+                        SignData::default()
+                    };
+                    let assigned_ecdsa = self.assign_ecdsa(&mut ctx, &chips, &signature)?;
+                    assigned_ecdsas.push(assigned_ecdsa);
+                }
+                Ok(())
+            },
+        )?;
+
+        let mut assigned_sig_verifs = Vec::new();
+        layouter.assign_region(
+            || "signature address verify",
+            |mut region| {
+                assigned_sig_verifs.clear();
+                // for i in 0..MAX_VERIF
+                for (i, assigned_ecdsa) in assigned_ecdsas.iter().enumerate() {
+                    let sign_data = signatures.get(i); // None when padding (enabled when address == 0)
+                    let offset = i;
+                    let (assigned_sig_verif, keccak_aux) = self.assign_signature_verify(
+                        config,
+                        &mut region,
+                        offset,
+                        randomness,
+                        &address_is_zero_chip,
+                        sign_data,
+                        assigned_ecdsa,
+                    )?;
+                    if i < signatures.len() {
+                        keccak_auxs.push(keccak_aux);
+                    }
+                    assigned_sig_verifs.push(assigned_sig_verif);
+                }
+
+                Ok(())
+            },
+        )?;
+
+        config.load_keccak(layouter, keccak_auxs, randomness)?;
```

Comment:
`i, // offset` ?

Reply:
I'm not sure what you're referring to here.

Reply:
Ok, seems that PR commit is not refreshed in the vscode plugin, I have to check this.

I'm referring to 

```
let offset = i;
...
let (assigned_sig_verif, keccak_aux) = self.assign_signature_verify(
    ... 
   offset
   ...
)
```

so you can use `i` , seems that is no need to create a new variable created `offset`here.